compiler/rustc_privacy/src/lib.rs

```diff
@@ -268,6 +268,7 @@ where
             | ty::FnPtr(..)
             | ty::Param(..)
             | ty::Error(_)
+            | ty::TyAlias(..)
```

Comment:
This is probably wrong, looks like this could allow using a private type alias to bypass privacy checks or something, likely needs explicit handling.

Reply:
This is certainly wrong, this visitor assumes that aliases are normalized away.

I'm kind of horrified by this PR's implications, looks like a potential source for numerous issues caused by forgetting to normalize aliases before using the type in random places across the compiler.